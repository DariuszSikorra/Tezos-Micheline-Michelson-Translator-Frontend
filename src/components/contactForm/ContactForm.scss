.contact-form {
    .contact-form_line {
        display: flex;
        justify-content: space-around;
    }

    .contact-form_button {
        font-size: 16px;
        margin-top: 15px;
        &:hover {
            cursor: pointer;
        }
    }
}