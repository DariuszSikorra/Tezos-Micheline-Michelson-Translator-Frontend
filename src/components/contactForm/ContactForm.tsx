import React, { useEffect } from "react";
import { useDispatch, useMappedState } from "redux-react-hook";
import { Formik } from "formik";
import * as Yup from "yup";
import { useGoogleReCaptcha } from "react-google-recaptcha-v3";
import FormInput from "../shared/input/FormInput";
import { FormValues } from "./types";
import "./ContactForm.scss";
import * as MessageTypes from "../../store/message/types";
import Alert from "../shared/alert/Alert";
import FormButton from "../shared/formButton/FormButton";

const validationSchema = Yup.object().shape({
  name: Yup.string(),
  group: Yup.string(),
  phone: Yup.number()
    .integer("It's a wrong phone number!")
    .min(8, "This phone number is too short."),
  email: Yup.string()
    .required("Email is required.")
    .email("It's a wrong email address."),
  content: Yup.string()
    .required("Message is required.")
    .min(10, "Message is too short."),
});

const mapState = (state: MessageTypes.IMessageGlobalState) => ({
  message: state.message,
});

const ContactForm = () => {
  const dispatch = useDispatch();
  const { message } = useMappedState(mapState);
  const { executeRecaptcha } = useGoogleReCaptcha();

  const submitForm = async (values: FormValues) => {
    if (!executeRecaptcha) return;

    const token = await executeRecaptcha("contact_form");
    if (!token.length) return;

    dispatch({
      type: MessageTypes.MESSAGE_SET,
      message: values,
    });

    dispatch({
      type: MessageTypes.MESSAGE_SEND,
      message: values,
      captcha: token,
    });
  };

  useEffect(() => {
    dispatch({
      type: MessageTypes.MESSAGE_SET_ERROR,
      isError: null,
    });
  }, [dispatch]);

  const saveForm = (values: FormValues) => {
    dispatch({
      type: MessageTypes.MESSAGE_SET,
      message: values,
    });
  };

  return (
    <div className="contact-form">
      <h2>Send message</h2>
      <Formik
        initialValues={{
          name: message.name,
          phone: message.phone,
          email: message.email,
          content: message.content,
        }}
        validationSchema={validationSchema}
        onSubmit={submitForm}
        validate={saveForm}
      >
        {({ values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting }) => (
          <form onSubmit={handleSubmit} noValidate>
            <div className="contact-form_line">
              <FormInput
                label="Name"
                type="text"
                name="name"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.name}
                errors={errors.name}
                touched={touched.name}
                className="contact-form_name"
              />
            </div>
            <div className="contact-form_line">
              <FormInput
                label="Phone"
                type="text"
                name="phone"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.phone}
                errors={errors.phone}
                touched={touched.phone}
              />
              <FormInput
                label="Email"
                type="email"
                name="email"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.email}
                errors={errors.email}
                touched={touched.email}
              />
            </div>
            <div className="contact-form_line">
              <FormInput
                label="How we can help you?"
                type="text"
                name="content"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.content}
                errors={errors.content}
                touched={touched.content}
                className="contact-form_message"
              />
            </div>
            <FormButton label="Submit" stylingType="submit" type="submit" disabled={isSubmitting} />
            {message.isError === false ? <Alert type="success" message="Message sent." /> : null}
          </form>
        )}
      </Formik>
    </div>
  );
};

export default ContactForm;
