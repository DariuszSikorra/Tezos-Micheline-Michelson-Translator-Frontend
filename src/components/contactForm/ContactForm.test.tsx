import React from "react";
import { mount, shallow } from "enzyme/build";
import { StoreContext } from "redux-react-hook";
import ContactForm from "./ContactForm";

import configureStore from "../../store/index";
import rootSaga from "../../store/saga";

const store = configureStore();
store.runSaga(rootSaga);

describe("ContactForm", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <StoreContext.Provider value={store}>
                <ContactForm setShowModal={() => { }} />
            </StoreContext.Provider>,
        );
    });

    it("matches snapshot", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});