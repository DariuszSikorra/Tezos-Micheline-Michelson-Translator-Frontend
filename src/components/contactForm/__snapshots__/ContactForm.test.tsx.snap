// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm matches snapshot 1`] = `
"<ContactForm setShowModal={[Function: setShowModal]}>
  <div className=\\"contact-form\\">
    <h2>
      Send message
    </h2>
    <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: submitForm]} validate={[Function: saveForm]} validateOnChange={true} validateOnBlur={true} isInitialValid={false} enableReinitialize={false}>
      <form onSubmit={[Function]} noValidate={true}>
        <div className=\\"contact-form_line\\">
          <FormInput label=\\"Name\\" type=\\"text\\" name=\\"name\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]} className=\\"contact-form_name\\">
            <div className=\\"form-input contact-form_name\\">
              <p className=\\"form-input_label\\">
                Name
              </p>
              <input name=\\"name\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Name\\" value=\\"\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
        </div>
        <div className=\\"contact-form_line\\">
          <FormInput label=\\"Phone\\" type=\\"text\\" name=\\"phone\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Phone
              </p>
              <input name=\\"phone\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Phone\\" value=\\"\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
          <FormInput label=\\"Email\\" type=\\"email\\" name=\\"email\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Email
              </p>
              <input name=\\"email\\" type=\\"email\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Email\\" value=\\"\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
        </div>
        <div className=\\"contact-form_line\\">
          <FormInput label=\\"How we can help you?\\" type=\\"text\\" name=\\"content\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]} className=\\"contact-form_message\\">
            <div className=\\"form-input contact-form_message\\">
              <p className=\\"form-input_label\\">
                How we can help you?
              </p>
              <input name=\\"content\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"How we can help you?\\" value=\\"\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
        </div>
        <button type=\\"submit\\" disabled={false} className=\\"contact-form_button\\">
          Submit
        </button>
      </form>
    </Formik>
  </div>
</ContactForm>"
`;
